lib/utils/azure_queues.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Implements utility functions for Azure Storage queues
+ */
+import * as winston from "winston";
+
+import { IContext } from "azure-functions-types";
+import { QueueService } from "azure-storage";
+
+import { Option, some } from "fp-ts/lib/Option";
+
+// Any delay must be less than 7 days (< 604800 seconds)
+const MAX_BACKOFF_MS = 7 * 24 * 3600 * 1000;
+const MIN_BACKOFF_MS = 285;
+
+// MAX_RETRIES *must* equal (maxDequeueCount - 1)
+// if (MAX_RETRIES < maxDequeueCount) - 1 there will be other extraneous
+//    #(maxDequeueCount - MAX_RETRIES) retries with the default visibilityTimeout
+//    before putting the message into the poison queue
+// if (MAX_RETRIES > maxDequeueCount - 1) the system will retry even after maxDequeueCount
+//    is reached and duplicate messages will be put into the poison queue
+export const MAX_RETRIES = Math.floor(
+  Math.log2(MAX_BACKOFF_MS / MIN_BACKOFF_MS)
+);
+
+/**
+ * Compute the timeout in seconds before the message will be processed again
```

Comment:
Add a note about the meaning of an empty option

Reply:
Also in the comment you're saying "seconds" whereas in the body you're using `MIN_BACKOFF_MS` without dividing by 1000 (an actual `Duration` type would be useful here) :)